stl/inc/algorithm

```diff
@@ -1389,6 +1389,14 @@ namespace ranges {
     template <input_iterator _It, sentinel_for<_It> _Se, weakly_incrementable _Out>
         requires indirectly_copyable<_It, _Out>
     _NODISCARD constexpr copy_result<_It, _Out> _Copy_unchecked(_It _First, const _Se _Last, _Out _Result) {
+        if constexpr (_Ptr_copy_cat<_It, _Out>::_Trivially_copyable) {
+            if (!_STD is_constant_evaluated()) {
+                auto _LastIt = _RANGES next(_First, _Last);

```

Comment:
Should we constraint this with `sized_sentinel_for` If this is really a trivially constructible type and you have to walk the whole range until `_Last` you are not really gaining anything

Reply:
I think that firstly walking the whole range to calculate the size and then `memmove`ing everything might be faster than walking and copying at the same time (it will probably also depend on the element size), but I haven't done any benchmarks.

Reply:
> walking the whole range to calculate the size and then `memmove`ing everything might be faster than walking and copying at the same time

It depends on the runtime behavior of the iterator and the iterator/sentinel comparison. When traversal requires examining the range contents - e.g., `transform_view`, or `take_while_view` - the extra traversal is generally going to be a loss if the entire range doesn't fit in first-level cache. I'd recommend the `sized_sentinel_for` constraint that @missco suggests. (I didn't do so in my review because there are some pre-existing cases in `<algorithm>` where an extra traversal let us reuse the `std` algorithm to get things done sooner. I've filed an issue to audit all uses of `ranges::next(i, s)` and `ranges::advance(i, s)` to look for places to clean up, and didn't mind this adding a place.)